### PR TITLE
Drop type specs for RecoMuon

### PR DESCRIPTION
**Pull request by NiharSaha**
#### PR description:
Drop type specs in Validation/RecoMuon python configuration files.

Central DQM migration campaign for drop type spces following
https://cms-sw.github.io/cms_coding_rules.html#6--packaging-rules-1
and update the safer syntax for existing parameters like: (1) drop type specifications where the original parameter exists. (2) using "clone" instead of "deepcopy" (3) move all parameters inside the clone

#### PR validation:
Tested in CMSSW_12_3_X via runTheMatrix.py -l limited -i all --ibeos

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1

File: Validation/RecoMuon/python/track_selectors_cff.py [hunk 1]
```diff
@@ -2,70 +2,79 @@
...
-probeTracks.minRapidity = cms.double(-2.4)
-probeTracks.maxRapidity = cms.double(2.4)
+probeTracks = PhysicsTools.RecoAlgos.recoTrackSelector_cfi.recoTrackSelector.clone(
+#quality = 'highPurity'   # previous setting
+#quality = ['loose']        # default
+    tip = 3.5,
+    lip = 30.,
+    ptMin = 3.0,
+    minRapidity = cms.double(-2.4),
+    maxRapidity = cms.double(2.4)
```
Comment (perrotta) [inline Validation/RecoMuon/python/track_selectors_cff.py:12]:
```suggestion
    minRapidity = -2.4,
    maxRapidity = 2.4
```

However, all these parameters are inside a `recoTrackSelectorPSet` : did you verify with an explicit dump whether they were actually modified? If so, then you should be also able to remove the explicit type from min/max rapidities

**Comment by qliphy** [comment]
ping @NiharSaha   on above comments

File: Validation/RecoMuon/python/track_selectors_cff.py (hunk 1 above)
Comment (NiharSaha) [reply Validation/RecoMuon/python/track_selectors_cff.py:12]: @perrotta Yes, I have checked that! Explicit types can be removed from min/max rapidities. Somehow it got missed from my side.

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**